src/config.rs

```diff
@@ -178,6 +178,108 @@ impl Display for Version {
     }
 }
 
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[cfg_attr(
+    feature = "python_bindings",
+    pyclass(frozen, get_all, module = "decomp_settings")
+)]
+/// Represents the set of important file and directory paths associated with a project version.
+///
+/// Each field corresponds to a specific artifact or directory relevant to decomp tools.
+pub struct VersionPaths {
+    /// Path to the original target binary (e.g., the ROM or executable to decompile). Usually called "baserom" by many projects.
+    ///
+    /// ## Examples
+    ///
+    /// ```yaml
+    /// target: "config/us/baserom_decompressed.us.z64"
+    /// ```
+    pub target: PathBuf,
+
+    /// Directory where build artifacts are generated.
+    ///
+    /// ## Examples
+    ///
+    /// ```yaml
+    /// build_dir: "build/ntsc-u/"
+    /// ```
+    pub build_dir: PathBuf,
+    /// Path to the map file generated during the build.
+    ///
+    /// ## Examples
+    ///
+    /// ```yaml
+    /// map: "build/us/drmario64.us.map"
+    /// ```
+    pub map: PathBuf,
+    /// Path to the binary produced by the project's build system.
+    ///
+    /// ## Examples
+    ///
+    /// ```yaml
+    /// compiled_target: "build/us/drmario64_uncompressed.us.z64"
+    /// ```
+    pub compiled_target: PathBuf,
+    /// Path to the intermediary ELF file generated during the build, if any.
+    ///
+    /// ## Examples
+    ///
+    /// ```yaml
+    /// elf: "build/pokemonsnap.elf"
+    /// ```
+    pub elf: Option<PathBuf>,
+
+    /// Directory containing the expected files used for comparison.
+    ///
+    /// Many projects simply put a copy of their `build` directory inside this expected directory.
+    ///
+    /// ## Examples
+    ///
+    /// ```yaml
+    /// expected_dir: "expected/"
+    /// ```
+    pub expected_dir: PathBuf,
+
+    /// Directory containing disassembled assembly files.
+    ///
+    /// ## Examples
+    ///
+    /// ```yaml
+    /// asm: "asm/"
+    /// ```
+    ///
+    /// ```yaml
+    /// asm: "asm/rev0/"
+    /// ```
+    pub asm: PathBuf,
+    /// Directory containing functions or files that have not yet been matched to the original binary.
+    ///
+    /// ## Examples
+    ///
+    /// ```yaml
+    /// nonmatchings: "asm/rev0/nonmatchings"
+    /// ```
+    pub nonmatchings: PathBuf,
```

Comment:
gc/wii projects don't use a nonmatchings dir, so I think this should be optional too, or we should have version-specific paths objects